plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -87,6 +87,46 @@ func (z *Zfs) gatherPoolStats(acc telegraf.Accumulator) (string, error) {
 	return strings.Join(pools, "::"), nil
 }
 
+func (z *Zfs) gatherDatasetStats(acc telegraf.Accumulator) (string, error) {
+	properties := []string{"name", "avail", "used", "usedsnap", "usedds"}
+
+	lines, err := z.zdataset(properties)
+	if err != nil {
+		return "", err
+	}
+
+	datasets := []string{}
+	for _, line := range lines {
+		col := strings.Split(line, "\t")
+
+		datasets = append(datasets, col[0])
+	}
+
+	if z.DatasetMetrics {
+		for _, line := range lines {
+			col := strings.Split(line, "\t")
+			if len(col) != len(properties) {
+				continue
```

Comment:
I still would like to see an INFO or WARN message here using the logging framework. Otherwise if e.g. the zfs tool changes the output we silently drop metrics. Is there a reason _NOT_ to add it?

Reply:
You can mark this as resolved.